internal/logger/logger.go

```diff
@@ -103,37 +124,43 @@ func Info(v ...interface{}) {
 
 type loggerFactory struct {
 	// If nil, log to stdout or stderr. Otherwise, log to this file.
-	file   *os.File
-	flag   int
-	format string
+	file      *os.File
+	sysWriter *syslog.Writer
+	flag      int
+	format    string
 }
 
 func (f *loggerFactory) newLogger(level, prefix string) *log.Logger {
 	return log.New(f.writer(level), prefix, f.flag)
 }
 
-func (f *loggerFactory) writer(level string) io.Writer {
-	if f.file != nil {
-		switch f.format {
-		case "json":
-			return &jsonWriter{
-				w:     f.file,
-				level: level,
-			}
-		case "text":
-			return &textWriter{
-				w:     f.file,
-				level: level,
-			}
+func (f *loggerFactory) createJsonOrTextWriter(level string, writer io.Writer) io.Writer {
+	if f.format == "json" {
+		return &jsonWriter{
+			w:     writer,
+			level: level,
+		}
+	} else {
```

Comment:
same here also. we can make the default as textWriter.
if f.format == "json" {
		return &jsonWriter{
			w:     writer,
			level: level,
		}
	}


return &textWriter {}